src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.cpp

```diff
@@ -43,293 +40,320 @@ DataType gauge_wave_deriv_h(const tnsr::I<DataType, 3>& x, const double t,
     : amplitude_(amplitude), wavelength_(wavelength) {
   ASSERT(amplitude > 0.0,
          "Amplitude must be non-negative. Given amplitude: " << amplitude_);
   ASSERT(wavelength > 0.0,
          "Wavelength must be non-negative. Given wavelength: " << wavelength_);
 }
 
-void GaugeWave::pup(PUP::er& p) noexcept {
+template <size_t Dim>
+void GaugeWave<Dim>::pup(PUP::er& p) noexcept {
   p | amplitude_;
   p | wavelength_;
 }
 
+template <size_t Dim>
 template <typename DataType>
-GaugeWave::IntermediateVars<DataType>::IntermediateVars(
+GaugeWave<Dim>::IntermediateVars<DataType>::IntermediateVars(
     const double amplitude, const double wavelength,
-    const tnsr::I<DataType, GaugeWave::volume_dim, Frame::Inertial>& x,
+    const tnsr::I<DataType, volume_dim, Frame::Inertial>& x,
     const double t) noexcept {
   h = gauge_wave_h(x, t, amplitude, wavelength);
   dx_h = gauge_wave_deriv_h(x, t, amplitude, wavelength);
   sqrt_h = sqrt(h);
   dx_h_over_2_sqrt_h = 0.5 * dx_h / sqrt(h);
 }
 
+template <size_t Dim>
 template <typename DataType>
-auto GaugeWave::variables(
-    const tnsr::I<DataType, GaugeWave::volume_dim, Frame::Inertial>& /*x*/,
-    const double /*t*/, const GaugeWave::IntermediateVars<DataType>& vars,
+auto GaugeWave<Dim>::variables(
+    const tnsr::I<DataType, volume_dim, Frame::Inertial>& /*x*/,
+    const double /*t*/, const IntermediateVars<DataType>& vars,
     tmpl::list<gr::Tags::Lapse<DataType>> /*meta*/) const noexcept
     -> tuples::TaggedTuple<gr::Tags::Lapse<DataType>> {
   return {Scalar<DataType>{vars.sqrt_h}};
 }
 
+template <size_t Dim>
 template <typename DataType>
-auto GaugeWave::variables(
-    const tnsr::I<DataType, GaugeWave::volume_dim, Frame::Inertial>& /*x*/,
-    const double /*t*/, const GaugeWave::IntermediateVars<DataType>& vars,
+auto GaugeWave<Dim>::variables(
+    const tnsr::I<DataType, volume_dim, Frame::Inertial>& /*x*/,
+    const double /*t*/, const IntermediateVars<DataType>& vars,
     tmpl::list<::Tags::dt<gr::Tags::Lapse<DataType>>> /*meta*/) const noexcept
     -> tuples::TaggedTuple<::Tags::dt<gr::Tags::Lapse<DataType>>> {
   return {Scalar<DataType>{-1.0 * vars.dx_h_over_2_sqrt_h}};
 }
 
+template <size_t Dim>
 template <typename DataType>
-auto GaugeWave::variables(
-    const tnsr::I<DataType, GaugeWave::volume_dim, Frame::Inertial>& x,
-    const double /*t*/, const GaugeWave::IntermediateVars<DataType>& vars,
+auto GaugeWave<Dim>::variables(
+    const tnsr::I<DataType, volume_dim, Frame::Inertial>& x, const double /*t*/,
+    const IntermediateVars<DataType>& vars,
     tmpl::list<DerivLapse<DataType>> /*meta*/) const noexcept
     -> tuples::TaggedTuple<DerivLapse<DataType>> {
   // most parts of d_lapse are zero, so make_with_value() here
-  auto d_lapse = make_with_value<
-      tnsr::i<DataType, GaugeWave::volume_dim, Frame::Inertial>>(x, 0.0);
+  auto d_lapse =
+      make_with_value<tnsr::i<DataType, volume_dim, Frame::Inertial>>(x, 0.0);
```

Comment:
Why does this use volume_dim instead of Dim?

Reply:
They are equivalent.  It used volume_dim before so I decided not to change it.  (And made new cases consistent with that choice.)